locations/spiders/iron_mountain.py

```diff
@@ -1,86 +1,18 @@
+class IronMountainSpider(scrapy.spiders.SitemapSpider):
     name = "iron_mountain"
     item_attributes = {
         "brand": "Iron Mountain Incorporated",
         "brand_wikidata": "Q1673079",
     }
-    start_urls = ("https://locations.ironmountain.com/",)
+    sitemap_urls = ["https://locations.ironmountain.com/robots.txt"]
+    download_delay = 1.0
 
     def parse(self, response):
-        urls = response.xpath(
-            '//*[@id="browse-expand"]/div/div[1]/div/li/div/a/@href'
-        ).extract()
-        for url in urls:
-            yield scrapy.Request(
-                url=url, callback=self.parse_list, meta={"country": "USA"}
-            )
-        urls = response.xpath(
-            '//*[@id="browse-expand"]/div/div[2]/div/li/div/a/@href'
-        ).extract()
-        for url in urls:
-            yield scrapy.Request(
-                url=url, callback=self.parse_list, meta={"country": "Canada"}
-            )
-
-    def parse_list(self, response):
-        urls = response.xpath(
-            '//*[@id="main-container"]/div[2]/div[3]/div[1]/div/div[2]/div[2]/ul/li/div/a/@href'
-        ).extract()
-        for url in urls:
-            yield scrapy.Request(
-                url=url,
-                callback=self.parse_city,
-                meta={"country": response.meta.get("country")},
-            )
-
-    def parse_city(self, response):
-        urls = response.xpath(
-            '//*[@id="main-container"]/div[2]/div[3]/div[1]/div/div[2]/div[2]/ul/li/a/@href'
-        ).extract()
-        for url in urls:
-            yield scrapy.Request(
-                url=url,
-                callback=self.parse_location,
-                meta={"country": response.meta.get("country")},
-            )
-
-    def parse_location(self, response):
-        data = re.sub(
-            "'", "", response.xpath('//*[@id="gmap"]/script/text()').extract()[0]
-        )
-        data = data.replace(",}", "}")
-        data = data.replace("RLS.defaultData = ", "")
-        data = data.replace(";", "")
-        address = json.loads(
-            "{"
-            + str.join(
-                ",",
-                json.loads(data)["markerData"][0]["info"]
-                .replace("</div>", "")
-                .split(",")[1:],
-            )
-        )
-        yield GeojsonPointItem(
-            addr_full=address["address_1"],
-            city=address["city"],
-            state=response.xpath(
-                '//*[@id="main-container"]/div[2]/div[1]/ol/li[2]/a/text()'
-            )
-            .extract()[0]
-            .strip(" "),
-            postcode=address["post_code"],
-            lat=float(address["lat"]),
-            lon=float(address["lng"]),
-            phone=address["local_phone"],
-            website=address["url"].replace("\\", ""),
-            country=response.meta.get("country"),
-            ref=address["url"].replace("\\", ""),
-        )
+        item = OpenGraphParser.parse(response)
+        if item["lat"]:
```

Comment:
I think this would error if lat isn't set. Did you mean .get?

Are there many pages in the sitemap without an lat? Should we do a sitemap filter instead?

Reply:
No to the first (look at DictParser logic)
No to the second.
And so no to the third.

I'll tell you what we want .... what we really really want:

item.has_position()

then we could save on these enjoyable discussions ... I believe I wrote that once somewhere.

Reply:
The DictParser will set "lat" to `None` if it can't find a latitude, so it should be good to check like this.